dds/idl/marshal_generator.cpp

```diff
@@ -2603,16 +2628,52 @@ namespace {
           }
           expr +=
             "  if (reached_end_of_struct) {\n" +
-            type_to_default("    ", type, stru_field_name, type->anonymous()) +
-            "  } else {\n"
-            "    if (!";
+            type_to_default("    ", type, stru_field_name, type->anonymous(), false, is_optional) +
+            "  } else {\n";
+          if (!is_optional) {
+            expr += "    if (!";
+          }
+        }
+
+        // TODO(tyler) This feels kind of hacky
+        // Can the optional branch be isolated completely
+        // Assigns strm value to temporary values
+        if (is_optional) {
+            //TODO(tyler) Is there an easier way to deal with strings here
+            AST_Type* const field_type = resolveActualType(field->field_type());
```

Comment:
Resolve warnings here: declaration of 'field_type' hides function parameter

Remove TODO comments

Check indent/formatting (looks like 2642 is more than 2 spaces to the right compared to 2641).